logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/OutgoingMessageRecipientsRetriever.kt

```diff
@@ -0,0 +1,91 @@
+package com.wire.kalium.logic.feature.message
+
+import com.wire.android.core.exception.Failure
+import com.wire.android.core.functional.Either
+import com.wire.android.core.functional.suspending
+import com.wire.android.feature.contact.DetailedContact
+import com.wire.android.feature.conversation.content.MessageRepository
+import com.wire.android.feature.conversation.data.ConversationRepository
+import com.wire.android.shared.prekey.PreKeyRepository
+import com.wire.android.shared.prekey.data.UserPreKeyInfo
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.feature.auth.AuthenticationResult
+import com.wire.kalium.logic.feature.contact.DetailedContact
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+
+class OutgoingMessageRecipientsRetriever(
+    private val preKeyRepository: PreKeyRepository,
+    private val conversationRepository: ConversationRepository,
+    private val messageRepository: MessageRepository
+) {
+
+    /**
+     * Gets detailed contacts for sending a new message in a conversation.
+     */
+    suspend fun prepareRecipientsForNewOutgoingMessage(
+        senderUserId: String,
+        conversationId: String
+    ): Either<AuthenticationResult.Failure, List<DetailedContact>> = suspending {
+        conversationRepository.detailedConversationMembers(conversationId).flatMap { detailedContacts ->
+            createSessionForMissingClientsIfNeeded(senderUserId, detailedContacts).map {
+                detailedContacts
+            }
+        }
+    }
+
+    private suspend fun createSessionForMissingClientsIfNeeded(
+        senderUserId: String,
+        detailedContacts: List<DetailedContact>
```

Comment:
I find it confusing to use both contact and user when referring to a "user".

Reply:
As I mentioned before, this `OutgoingMessageRecipientsRetriever` actually shouldn't be in this PR yet, I let it leak by accident.

I want to review these names and alter some stuff inside of it when the time to submit it comes.

My suggestion is something along these lines:

|Term | Description|
|----|---|
| `User` | Any user, generic. Can be "self" user. _i.e._ the user that is currently operating this client, or any other when in context of searching for connections and vieweing other profiles, for example. |
| `Contact` | Another user that this self-user has a conversation with. Used in context of conversations. Maybe can be replaced by `ConversationMember`? |
| `Recipient` | Grouping of `Contact`/`ConversationMember` + its Client IDs. Used in context of creating outgoing messages / sending messages, encrypting, creating an envelope in general |


This `DetailedContact` would be replaced with `Recipient` in this case. The `senderUserId` won't be needed at all for this operation, as it is injected via the `sessionScope`.